lms/views/exceptions.py

```diff
@@ -60,6 +60,20 @@ def reused_consumer_key(self):
 
         return {}
 
+    @exception_view_config(
+        context=SerializableError, renderer="lms:templates/error_dialog.html.jinja2"
+    )
+    def serializable_error(self):
+        self.request.response.status_int = 400
+
+        self.request.context.js_config.enable_error_dialog_mode(
+            error_code=self.exception.error_code,
+            message=self.exception.message,
+            error_details=self.exception.details,
+        )
+
+        return {}
```

Comment:
If you look at the `ReusedConsumerKey` case here it could be completely covered by turning that into a child of `SerializableError`.